make set-unstable-versions fails with errors

**Bug description**
Unable to run `make set-unstable-versions` which blocks plugin development (like TCE). Errors with the following:

```
[vonthd@centos8 tanzu-framework]$ make set-unstable-versions
which: no controller-gen in (/home/vonthd/.gvm/pkgsets/go1.16.4/global/bin:/home/vonthd/.gvm/gos/go1.16.4/bin:/home/vonthd/.gvm/pkgsets/go1.16.4/global/overlay/bin:/home/vonthd/.gvm/bin:/home/vonthd/.gvm/bin:/home/vonthd/.nvm/versions/node/v10.24.1/bin:/home/vonthd/.gvm/pkgsets/go1.16.4/global/bin:/home/vonthd/.gvm/gos/go1.16.4/bin:/home/vonthd/.gvm/pkgsets/go1.16.4/global/overlay/bin:/home/vonthd/.gvm/bin:/home/vonthd/.gvm/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/vonthd/go/bin:/home/vonthd/.local/bin:/home/vonthd/bin:/home/vonthd/go/bin)
TANZU_CLI_NO_INIT=true GOPRIVATE="github.com/vmware-tanzu" go run -ldflags "-s -w -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildDate=$(date -u +"%Y-%m-%d")' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildSHA=$(git describe --match= --always --dirty)' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildVersion=v1.4.0-pre-alpha-2' -X 'main.BuildEdition=tkg' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/buildinfo.IsOfficialBuild=""'" ./cmd/cli/tanzu/main.go config set unstable-versions "experimental"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb51412]

goroutine 1 [running]:
github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1.(*ClientConfig).SetUnstableVersionSelector(...)
	/home/vonthd/go/src/github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1/clientconfig.go:52
github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/core.glob..func4(0x129b7e0, 0xc0003cbfe0, 0x1, 0x1, 0x0, 0x0)
	/home/vonthd/go/src/github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/core/config.go:110 +0xb2
github.com/spf13/cobra.(*Command).execute(0x129b7e0, 0xc0003cbfb0, 0x1, 0x1, 0x129b7e0, 0xc0003cbfb0)
	/home/vonthd/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x129a8e0, 0x0, 0x0, 0x4424fc)
	/home/vonthd/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/vonthd/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/core.Execute(0xc000000180, 0xc000060740)
	/home/vonthd/go/src/github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/core/root.go:130 +0x50
main.main()
	/home/vonthd/go/src/github.com/vmware-tanzu/tanzu-framework/cmd/cli/tanzu/main.go:15 +0x26
exit status 2
make: *** [Makefile:299: set-unstable-versions] Error 1
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] Installation
- [x] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
It should not error out

**Steps to reproduce the bug**
run `make set-unstable-versions`

**Version** (include the SHA if the version is not obvious)
using last main

--- a/apis/config/v1alpha1/clientconfig.go
+++ b/apis/config/v1alpha1/clientconfig.go
@@ -47,6 +47,12 @@ func (c *ClientConfig) GetCurrentServer() (*Server, error) {
 // experimental: all pre-release versions without +build semver data
 // all: return all unstable versions.
 func (c *ClientConfig) SetUnstableVersionSelector(f VersionSelectorLevel) {
+	if c.ClientOptions == nil {
+		c.ClientOptions = &ClientOptions{}
+	}
+	if c.ClientOptions.CLI == nil {
+		c.ClientOptions.CLI = &CLIOptions{}
+	}
 	switch f {
 	case AllUnstableVersions, AlphaUnstableVersions, ExperimentalUnstableVersions, NoUnstableVersions:
 		c.ClientOptions.CLI.UnstableVersionSelector = f

